eosmetrics/emtr-aggregate-timer.c

```diff
@@ -0,0 +1,163 @@
+/* -*- mode: C; c-file-style: "gnu"; indent-tabs-mode: nil; -*- */
+
+/* Copyright 2021 Endless OS Foundation, LLC. */
+
+/* This file is part of eos-metrics.
+ *
+ * eos-metrics is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published
+ * by the Free Software Foundation, either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * eos-metrics is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with eos-metrics.  If not, see
+ * <http://www.gnu.org/licenses/>.
+ */
+
+#include "emtr-aggregate-timer-private.h"
+
+
+/**
+ * SECTION:emtr-aggregate-timer
+ * @title: Aggregate events timer
+ * @short_description: Timer to aggregate events.
+ * @include: eosmetrics/eos-aggregate-timer.h
+ *
+ * EmtrAggregateTimer is a simple timer to aggregate events.
+ */
+
+struct _EmtrAggregateTimer
+{
+  GObject parent_instance;
+
+  GCancellable *cancellable; /* (owned) */
+  EmerAggregateTimer *timer_proxy; /* (owned) */
+  EmerEventRecorderServer *dbus_proxy; /* (unowned) */
+
+  gboolean stopped;
+};
+
+G_DEFINE_TYPE (EmtrAggregateTimer, emtr_aggregate_timer, G_TYPE_OBJECT)
+
+static void
+emtr_aggregate_timer_finalize (GObject *object)
+{
+  EmtrAggregateTimer *self = (EmtrAggregateTimer *)object;
+
+  g_cancellable_cancel (self->cancellable);
+  g_clear_object (&self->cancellable);
+
+  if (!self->stopped && self->timer_proxy)
+    {
+      g_autoptr (GError) error = NULL;
+      emer_aggregate_timer_call_stop_timer_sync (self->timer_proxy, NULL, &error);
```

Comment:
Since we are ignoring the result anyway:

```suggestion
      emer_aggregate_timer_call_stop_timer (self->timer_proxy, NULL, NULL, NULL);
```

No need to block.